modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/model.scala

```diff
@@ -222,7 +257,12 @@ object Queue {
 /**
   * A list of Steps grouped by target and instrument.
   */
-case class Sequence[A](id: String, steps: NonEmptyList[Step[A]])
+case class Sequence[A](
+  id: String,
+  steps: NonEmptyList[Step[A]],
+  // Unrooted executions
```

Comment:
Yes, those structures are equivalent.

The thing is that with in the Engine we are all the time working with the *current* execution. This structure with 2 queues and a cursor resembles a [`Zipper`](https://en.wikipedia.org/wiki/Zipper_(data_structure)) which makes the focused element (the current execution) very accessible and facilitates moving back and forth.

Without a `Zipper` we would have to do the work of finding where we are each time we want to change the State.

In any case, a `Zipper` is expected to be *zipped* and *unzipped* from and to its original structure which in this case will be a whole single tree with each element having its own status flag.